src/samples/copypaste-socks-chromeapp/freedom-module.ts

```diff
@@ -131,6 +140,46 @@ freedom().on('handleSignalMessage', (signal:WebRtc.SignallingMessage) => {
   }
 });
 
+// Crypto helper functions
+function ab2str(buf:ArrayBuffer) {
+  return String.fromCharCode.apply(null, new Uint16Array(buf));
+}
+
+function str2ab(str:string) {
+  var buf = new ArrayBuffer(str.length * 2);  // 2 bytes for each char
+  var bufView = new Uint16Array(buf);
+  for (var i = 0, strLen = str.length; i < strLen; i++) {
+    bufView[i] = str.charCodeAt(i);
+  }
+  return buf;
+}
+
+// Crypto request messages
+freedom().on('friendKey', (newFriendKey:string) => {
+  friendKey = newFriendKey;
+});
+
+freedom().on('signEncrypt', (message:string) => {
+  pgp.signEncrypt(str2ab(message), friendKey, true)
+    .then(function (cipherdata:ArrayBuffer) {
```

Comment:
Actually, this probably shouldn't use `function` at all.  In general we prefer the fat-arrow (`=>`) functions, which are automatically bound.  This makes `this` a lot less confusing.